linkingExamples_test.go

```diff
@@ -0,0 +1,133 @@
+// or some network storage, or... whatever you want, really :)
+var store = storage.Memory{}
+
+// TODO: These examples are really heavy on CIDs and the multicodec and multihash magic tables.
+// It would be good to have examples that create and use less magical LinkSystem constructions, too.
+
+func ExampleStoringLink() {
+	// Creating a Link is done by choosing a concrete link implementation (typically, CID),
+	//  getting a LinkSystem that knows how to work with that, and then using the LinkSystem methods.
+
+	// Let's get a LinkSystem.  We're going to be working with CID links,
+	//  so let's get the default LinkSystem that's ready to work with those.
+	lsys := cidlink.DefaultLinkSystem()
+
+	// We want to store the serialized data somewhere.
+	//  We'll use an in-memory store for this.  (It's a package scoped variable.)
+	//  You can use any kind of storage system here;
+	//   you just need a function that conforms to the ipld.BlockWriteOpener interface.
+	lsys.StorageWriteOpener = (&store).OpenWrite
+
+	// To create any links, first we need a LinkPrototype.
+	// This gathers together any parameters that might be needed when making a link.
+	// (For CIDs, the version, the codec, and the multihash type are all parameters we'll need.)
+	// Often, you can probably make this a constant for your whole application.
+	lp := cidlink.LinkPrototype{cid.Prefix{
+		Version:  1,    // Usually '1'.
+		Codec:    0x71, // 0x71 means "dag-cbor" -- See the multicodecs table: https://github.com/multiformats/multicodec/
+		MhType:   0x15, // 0x15 means "sha3-384" -- See the multicodecs table: https://github.com/multiformats/multicodec/
+		MhLength: 48,   // sha3-384 hash has a 48-byte sum.
+	}}
+
+	// And we need some data to link to!  Here's a quick piece of example data:
+	n := fluent.MustBuildMap(basicnode.Prototype.Map, 1, func(na fluent.MapAssembler) {
+		na.AssembleEntry("hello").AssignString("world")
+	})
+
+	// Before we use the LinkService, NOTE:
+	//  There's a side-effecting import at the top of the file.  It's for the dag-cbor codec.
+	//  The CID LinkSystem defaults use a global registry called the multicodec table;
+	//  and the multicodec table is populated in part by the dag-cbor package when it's first imported.
+	// You'll need that side-effecting import, too, to copy this example.
+	//  It can happen anywhere in your program; once, in any package, is enough.
+	//  If you don't have this import, the codec will not be registered in the multicodec registry,
+	//  and when you use the LinkSystem we got from the cidlink package, it will return an error of type ErrLinkingSetup.
+	// If you initialize a custom LinkSystem, you can control this more directly;
+	//  these registry systems are only here as defaults.
+
+	// Now: time to apply the LinkSystem, and do the actual store operation!
+	lnk, err := lsys.Store(
+		ipld.LinkContext{}, // The zero value is fine.  Configure it it you want cancellability or other features.
```

Comment:
Not sure how I feel about the simple cases having to type this out :) I get that it's the case with `context.Context` too, but in most cases you do end up having a `ctx` parameter or variable you can easily reuse. Not really the case with a custom `ipld.LinkContext`.

Reply:
Yeah, I'd like to be able to get rid of this.  I think we'd need more methods for it, though?  If you have naming suggestions, I'd be happy to add them.

We can also add more methods later.